### PR TITLE
fix #269845 Svg don't inherit QPaintEnginePrivate 

### PR DESCRIPTION
**Pull request by ericfont**
SvgPaintEnginePrivate class has always extended QPaintEnginePrivate.  But QPaintEnginePrivate was defined in a header file which was copy-and-pasted from Qt over 6 years ago in the initial git commit.  Apparently as of 5.10, Qt has changed the definition of QPaintEnginePrivate, and that has caused a segfault whenever I tried to run mscore 2.1, 2.2, or master on my armv7 arch linux machine when using Qt 5.10 (although worked with Qt 5.9).

The fix here is to build SvgPaintEnginePrivte using the version of QPaintEnginePrivate from the installed Qt version's header file found by CMakeLists, instead of using the copy-and-pasted header file, which I'm deleting.

### CODE REVIEW

**Comment by ericfont** [comment]
note this should go into both 2.2 and master.

File: mscore/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ include_directories(
       ${PROJECT_SOURCE_DIR}
       ${PROJECT_SOURCE_DIR}/thirdparty
       ${QTSINGLEAPPLICATION_INCLUDE_DIRS}
+      ${Qt5Gui_PRIVATE_INCLUDE_DIRS}
```
Comment (ericfont) [inline mscore/CMakeLists.txt]: I don't know if this is the most appropriate place to add this include.

**Comment by Jojo-Schmitz** [comment]
As MuseScore 2.2 is supposed to be using Qt-5.4, and master to currently use Qt 5.9, I'm not sure it is really needed for 2.2?

**Comment by ericfont** [comment]
No, it is needed for 2.2, because users who have never versions of Qt 5 installed on their systems will still need to be able to run musescore.  MuseScore packages on bleeding-edge distros will fail (not everyone will use AppImages or be able to downgrade their Qt installation).

Anyway, it is bad practice to copy and paste that private header.

**Comment by Jojo-Schmitz** [comment]
Hmm, isn't it equally bad practice using private headers?

**Comment by ericfont** [comment]
Of course using private headers is bad practice, but using a private header which is copy-and-pasted from many years and not up-to-date is worse.

I'd be happy if there was a way to modify SvgPaintEngine so that it doesn't extend QPaintEnginePrivate, but I imagine that would be much more involved.

**Comment by ericfont** [comment]
I've traced down what I believe is Qt's particular commit "Fix clipping of graphics effects in high dpi mode" which resulted in the modification to the header file, whereby an additional QRegion "baseSystemClip" was added to the QPaintEnginePrivate class: https://github.com/qt/qtbase/commit/3398d9d40cb0dae2dc2a1a4f7dc3b4b9cceae903

Bleeding-edge distros (and self-builders) which use newer Qt 5 versions will benefit from this Qt bug fix, so I think that is an argument for not copy-and-pasting an out-of-date header.

**Comment by lasconic** [comment]
On Mac, Windows and AppImage, this should be transparent since the header will match the version embedded with MuseScore packages but for Linux distrib, I'm not sure if it really solves the problem. My understanding is that MuseScore will need to be run with the same (binary compatible) version of Qt that was used to build it.

**Comment by ericfont** [comment]
I modified the commit message for clarity.  Previously I titled it "use Qt private header, don't copy-paste" which was a bit misleading because both before and after this commit a private header is still used, but the main difference is that I'm getting the header from the installed Qt version found by CMake instead of using the copy-pasted file.

**Comment by ericfont** [comment]
@lasconic wondered on IRC to maybe remove the inheritance altogether.  I've tried that and found that MuseScore is able to compile and run.  I like this solution better because it avoids needed to rely on any private header.  So I've updated this PR accordingly.

**Comment by lasconic** [comment]
Please also test if SVG can be exported :)

**Comment by ericfont** [comment]
Cross-linking to issue https://musescore.org/en/node/257551 "Embeds Qt5's private headers without putting them in a namespace" since the above PR will fix that debian bug by entirely removing the private header inheritance entirely.

**Comment by ericfont** [comment]
I applied the corrected constructor @lasconic told me, and the PR correctly exports SVG in 2.2, even when running in QtCreator.

**Pull request merged**